utils/docker-utils.sh

```diff
@@ -132,6 +139,66 @@ docker::prune_configs() {
     docker config rm $(docker config ls -qf label=name="$CONFIG_LABEL") &>/dev/null
 }
 
+# Deploy a service, it will set config digests (in case a config is defined in the compose file)
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring" ...
+# - $2 : docker compose file, e.g. "docker-compose.yml" "docker-compose.cluster.yml" ...
+# - $3 : docker compose dev file, e.g. "docker-compose.dev.yml"
+# - $4 : services names, e.g. "monitoring" "hapi-fhir" ...
+#
+docker::deploy_service() {
+    local -r DOCKER_COMPOSE_PATH="${1:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_FILE="${2:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_DEV_FILE="${3:?"FATAL: function 'deploy_service' is missing a parameter"}"
+    local -r SERVICE_NAMES="${4:?"FATAL: function 'deploy_service' is missing a parameter"}"
+
+    # Check for need to set config digests
+    local -r files=($(yq '.configs."*.*".file' "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"))
+    if [ "${files[*]}" != "null" ]; then
+        config::set_config_digests "${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE"
+    fi
+
+    try "docker stack deploy -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_FILE -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE instant"
+    docker::deploy_sanity "${SERVICE_NAMES[@]}"
+}
+
+# Deploy a config importer:
+# Sets the config digests, deploy the config importer, remove it and remove the stale configs
+#
+# Arguments:
+# - $1 : docker compose path, e.g. "/instant/monitoring" ...
+# - $2 : services name, e.g. "clickhouse-config-importer" ...
+# - $3 : config label, e.g. "clickhouse" "kibana" ...
+# - $4 : docker compose file name, default "docker-compose.config.yml"
+#
+docker::deploy_config_importer() {
+    local -r CONFIG_COMPOSE_PATH="${1:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r SERVICE_NAME="${2:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r CONFIG_LABEL="${3:?"FATAL: function 'deploy_config_importer' is missing a parameter"}"
+    local -r DOCKER_COMPOSE_FILE="${4:-"docker-compose.config.yml"}"
```

Comment:
This default is a possible source of bugs, let's make the implementer be specific about specifying the path to the file. You can do this by merging `CONFIG_COMPOSE_PATH` and `DOCKER_COMPOSE_FILE` into one variable. Then you just make sure this function throws an error if the file doesn't exist.

Reply:
done in [e5e5ce8](https://github.com/jembi/platform/pull/188/commits/e5e5ce8efc4d8efd46fecbad8bc7ccf535d6e8d5)